src/content-script/worker.ts

```diff
@@ -93,10 +96,12 @@ export class Worker {
       const scheduleAppointment = new Scheduler(params, userVisit);
       const res = await scheduleAppointment.handle(task);
       if (res.isDone) {
-        await this.stop();
-      } else {
-        this.settingAppointment = false;
+        await this.stop(res.status);
+      } else if (res.status) {
+        dispatchSearchStatus(res.status)
```

Comment:
Is this related to the last if? if not, it should be IMO with `else`, WDUT?

Reply:
No need to change the status if `res.status` doesn't indicate a new status (success/"already has apt."/"trying again")